### ISSUE TITLE
Questions on contrastive training

### ISSUE THREAD

**Issue by xyimaging** [dewenzeng/positional_cl#5]
Hi, I'm trying to train the contrastive encoder from scratch on the ACDC data set, however, unlike your result shown on #3,  my loss is stuck at 5.5597 from epoch 0 to epoch 30 until I cancelled it due to long time running. I'm wondering if I missed any tricks?

My results:
1. looks fine on the first several iterations
<img width="323" alt="截屏2022-03-01 下午3 27 47" src="https://user-images.githubusercontent.com/56007678/156251803-2423244e-176e-43a1-ab73-777c05b8c820.png">

2. stucked into 5.5597 in the following 30 epochs
<img width="360" alt="截屏2022-03-01 下午3 29 16" src="https://user-images.githubusercontent.com/56007678/156251992-ce104a84-d021-4919-906e-4231934844f0.png">
<img width="362" alt="截屏2022-03-01 下午3 29 33" src="https://user-images.githubusercontent.com/56007678/156252016-9687f029-d81f-41c8-9cf8-612c2043e821.png">

I run the code with:
```
python3 train_contrast.py --device cuda:0 --batch_size 25 --epochs 300 --data_dir "./dataset/02_08_2022/acdc/unlabeled/" --lr 0.1 --do_contrast --dataset acdc --patch_size 352 352 \
--experiment_name contrast_acdc_pcl_temp01_thresh035 --slice_threshold 0.35 --temp 0.1 --initial_filter_size 48 --classes 512 --contrastive_method pcl
```
Here I set the batch_size to 25 to avoid the memory crush.

**Comment by dewenzeng** [comment]
@xyimaging On my end, when using batch size 24, it works fine. Maybe it's because of the data you are using, when I use bs=24, the total number of batches is 1056, which is different from yours.
![image](https://user-images.githubusercontent.com/75713682/156272944-bc24e970-e3c0-4925-bb35-6a0726bb20f8.png)
the training loss curve with bs=24
![image](https://user-images.githubusercontent.com/75713682/156273106-ca385abe-0c95-4907-99d8-0b92fc2f18ab.png)

One thing you may also try is the learning rate warmup https://github.com/HobbitLong/SupContrast/blob/a8a275b3a8b9b9bdc9c527f199d5b9be58148543/main_supcon.py#L216

**Comment by xyimaging** [comment]
The bug is solved, thanks!
